pydis_site/apps/content/resources/guides/python-guides/subclassing_context.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Subclassing Context
+description: "Subclassing the default `commands.Context` class to add more functionability and customisability."
+---
+
+Start by reading the guide on [subclassing the `Bot` class](./subclassing_bot.md). A subclass of Bot has to be used to
+inject your custom context subclass into discord.py.
+
+## Overview
+
+The way this works is by creating a subclass of discord.py's [`Context` class](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Context)
+adding whatever functionality you wish. Usually this is adding custom methods or properties, so that you don't need to
+copy it around or awkwardly import it elsewhere.
+
+This guide will show you how to add a `prompt()` method to the context and how to use it in a command.
+
+## Example subclass and code
+
+The first part - of course - is creating the actual context subclass. This is done similarly to creating a bot
+subclass, it will look like this:
+
+```python
+from typing import Optional
+
+from discord import RawReactionActionEvent
+from discord.ext import commands
+
+
+class CustomContext(commands.Context):
+    async def prompt(
+            self,
+            message: str,
+            *,
+            timeout=30.0,
+            delete_after=True
+    ) -> Optional[bool]:
+        """Prompt the author with an interactive confirmation message.
+
+        This method will send the `message` content, and wait for max `timeout` seconds
+        (default is `30`) for the author to react to the message.
+
+        If `delete_after` is `True`, the message will be deleted before returning a
+        boolean or None (if the author didn't respond) indicating whether the author
+        confirmed or denied.
+        """
+        msg = await self.send(message)
+
+        for reaction in ('\N{WHITE HEAVY CHECK MARK}', '\N{CROSS MARK}'):
```

Comment:
I believe this should work and looks better.
```suggestion
        for reaction in ('✅', '❌'):
```

Reply:
This was a conscious decision, as it is generally seen as bad practice to store unicode characters in source files: they are quite difficult to work with. The alternative used here is the most readable one.

That said, I will change this since it appears that I have been outvoted.